src/importexport/musicxml/internal/musicxml/exportxml.cpp

```diff
@@ -4850,6 +4851,47 @@ void ExportMusicXml::words(TextBase const* const text, staff_idx_t staff)
 
     directionTag(m_xml, m_attr, text);
     wordsMetronome(m_xml, m_score->style(), text, offset);
+
+    directionETag(m_xml, staff);
+}
+
+//---------------------------------------------------------
+//   systemText
+//---------------------------------------------------------
+
+void ExportMusicXml::systemText(StaffTextBase const* const text, staff_idx_t staff)
+{
+    const auto offset = calculateTimeDeltaInDivisions(text->tick(), tick(), m_div);
+
+    if (text->plainText() == "") {
+        // sometimes empty Texts are present, exporting would result
+        // in invalid MusicXML (as an empty direction-type would be created)
+        return;
+    }
+
+    directionTag(m_xml, m_attr, text);
+    wordsMetronome(m_xml, m_score->style(), text, offset);
+
+    if (text->swing()) {
+        m_xml.startElement("sound");
+        m_xml.startElement("swing");
+        if (!text->swingParameters().swingUnit) {
+            m_xml.tag("straight");
+        } else {
+            const int swingPercentage = text->swingParameters().swingRatio;
+            const int swingDivisor = std::gcd(text->swingParameters().swingRatio, 100);
+            m_xml.tag("first",  100 / swingDivisor);
+            m_xml.tag("second", swingPercentage / swingDivisor);
+            if (text->swingParameters().swingUnit == Constants::DIVISION / 2) {
+                m_xml.tag("swing-type", TConv::toXml(DurationType::V_EIGHTH));
+            } else {
+                m_xml.tag("swing-type", TConv::toXml(DurationType::V_16TH));
+            }
```

Comment:
why not ```cpp
            } else if (text->swingParameters().swingUnit == Constants::DIVISION / 4){
                m_xml.tag("swing-type", TConv::toXml(DurationType::V_16TH));
            } else {
                m_xml.tag("swing-type", TConv::toXml(DurationType::V_ZERO));
            }
```

Reply:
Because `V_ZERO` case is handled before (and would result in invalid MusicXML anyway).